### PR TITLE
Add compatibility with PHP 8.0

### PR DESCRIPTION
**Pull request by ausi**
As discussed in #66

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,22 +10,21 @@ cache:
...
+    - php: 7.3
     - php: 7.2
       if: branch = master AND fork = false
       env:
         - UPDATE_RESOURCES=true
         - secure: "KdRIoeVMAjv63erXmG3EkOnE/UrIsgIJ7/S3yZwJq3y+c6jd7Xax6lNWRF6PcfKtNHJNhreIKA/skW5CkPktKIKqOrHrJVx5GsIQX3OqW1z2LF/qqDr7zxNWWrCCI13t7uy4GkPNU0xMwauk+5GCoyBrsxwjwf3/Lm3SA7xZTrE="
+    - php: nightly
+      install:
+        - composer self-update --preview
+        - composer install --ignore-platform-req=php
```
Comment (ausi) [inline .travis.yml:27]: This is required to ignore dependency incompatibilities with PHP 8.0. This should be remove once all dependencies are compatible with PHP 8.

**Comment by ausi** [comment]
The tests run through now for PHP 8.0.

The psalm errors seem to be the same as for PHP 7 so they are probably not related to this pull request.

**Comment by lstrojny** [comment]
Thank you!

**Comment by hexus** [comment]
I feel like making this change as part of a patch release is a breaking one. Compatibility jumped from `>=5.3` to `7.2 || 8.0`.

**Comment by ausi** [comment]
PHP 7.2 was required since version 3.9.8 of uap-php, this is not related to this pull request. I think you should create a seperate issue for that.

**Comment by hexus** [comment]
Yes, it's #58 instead, my bad.

**Pull request merged**